### PR TITLE
Fix issue with resizing terminal.

### PR DESCRIPTION
**Pull request by amejia1**
Need to ensure the active buffer is always being used.
Fixes https://github.com/xtermjs/xterm.js/issues/824 .

### CODE REVIEW

**Comment by amejia1** [comment]
As far as I know, the change here will fix the issue with resizing the terminal as well as ensure no other problems arise and continues to pass the test suite. Works for me on Windows and Linux.

File: src/SelectionManager.ts [hunk 1]
```diff
@@ -139,16 +142,6 @@ export class SelectionManager extends EventEmitter implements ISelectionManager
...
   }
 
-  /**
-   * Sets the active buffer, this should be called when the alt buffer is
-   * switched in or out.
-   * @param buffer The active buffer.
-   */
-  public setBuffer(buffer: IBuffer): void {
-    this._buffer = buffer;
-    this.clearSelection();
```
Comment (Tyriar) [inline src/SelectionManager.ts]: It looks like selection isn't going to be cleared when the terminal buffer switches, which feels awkward as random text is then selected.

Comment (amejia1) [reply src/SelectionManager.ts]: Alright, added a new commit to fix this.

**Comment by Tyriar** [comment]
I made a few changes in https://github.com/xtermjs/xterm.js/pull/1206/commits/b5aff0be7e9b7af13c3488b0c6f05e54a9c5c24b which I think are correct, @amejia1 please double check they're good then we can push this in.

**Comment by amejia1** [comment]
New changes fail the test suite. Just check on my Windows box. The Travis CI tests are already failing too.

**Comment by amejia1** [comment]
So yes, the old BufferSet can't be kept when resetting the terminal, otherwise the test suite fails and it doesn't actually clear anything when actually resetting a terminal. Also, since the BufferSet has to be changed, the selection manager has to be cleared, otherwise it throws errors about not being able to read properties of `undefined`.

File: src/Terminal.ts [hunk 2]
```diff
@@ -292,7 +292,6 @@ export class Terminal extends EventEmitter implements ITerminal, IInputHandlingT
 
     // Create the terminal's buffers and set the current buffer
     this.buffers = new BufferSet(this);
-    this.emit('setup');
```
Comment (amejia1) [inline src/Terminal.ts]: There's going to be a need for some kind of notification when the BufferSet changes. Best way I could think of was to use the event emitter. Should I reimplement this? Maybe indicate it as a private event with a name like '_changebufferset'?

File: src/Terminal.ts [hunk 3]
```diff
@@ -2073,9 +2072,11 @@ export class Terminal extends EventEmitter implements ITerminal, IInputHandlingT
     this.options.cols = this.cols;
     const customKeyEventHandler = this.customKeyEventHandler;
     const inputHandler = this.inputHandler;
+    const buffers = this.buffers;
```
Comment (amejia1) [inline src/Terminal.ts]: Was trying to mention this earlier. This causes the test suite to fail.

**Comment by Tyriar** [comment]
> Ah. This causes the test suite to fail. The old BufferSet can't be kept otherwise resetting the terminal won't do the right thing.

@amejia1 I was a little confused by this, do you know why `term.reset()` works (clears the buffer) when the `BufferSet` is being reset? I think @parisk did that part.

screen shot 2018-01-16 at 12 07 41 pm

> Should this clear selection when the terminal resets?

Oh it wasn't clear that's what this was doing. Maybe for now let's just have a specific call to SelectionManager in reset?

**Comment by amejia1** [comment]
@Tyriar Ok, so clearing the selection manager on reset turns out to be much cleaner. This passes the test suite and I ensured that resetting the terminal does clear the terminal and also any pending selection.

**Comment by Tyriar** [comment]
Thanks heaps @amejia1, sorry about the delay 😃

**Pull request merged**